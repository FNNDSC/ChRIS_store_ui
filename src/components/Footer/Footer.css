.footer {
  /* color */
  background: var(--pf-black-900);
  background: linear-gradient(var(--pf-black-900), var(--pf-black));
  color: var(--pf-black-100);

  /* display */
  padding: 4em 1em;
  padding-bottom: 8em;

  /* position */
  position: relative;

  /* font */
  font-size: 1.4em;
}

.footer-row {
  /* display */
  width: 100%;
  max-width: 1500px;
  margin: 0 auto !important;
}

.footer-header {
  /* display */
  padding: 0.4em 1em;

  /* font */
  font-size: 1.3em;
  line-height: 1.3;
  white-space: pre-line;
}

.footer-header > strong { font-size: 1.2em; }

.footer-logo {
  /* display */
  padding-bottom: 2em;

  /* flex */
  flex: 1;
  flex-shrink: 0;
}

.footer-logo > img {
  /* display */
  width: 95%;
  max-width: 358.8px; /* img initial width (598px) * (3/5) */
  height: auto;
}

.footer-body {
  /* flex */
  flex: 2;
}

.footer-desc { font-weight: 300; }

.footer-links {
  /* display */
  margin-top: 2em;

  /* font */
  text-align: center;
}

.footer-link {
  /* display */
  padding-top: 0.3em;

  /* font */
  font-weight: 600;
}

.footer-copyright {
  /* position */
  position: absolute;
  bottom: 1em;
  left: 0;
  right: 0;

  /* font */
  text-align: center;
}

/* ============================== */
/* ---------- RESPONSIVE -------- */
/* ============================== */

@media (min-width: 480px) {
  .footer-header {
    /* display */
    padding: 0.35em 0.8em;

    /* font */
    font-size: 1.4em;
    line-height: 1.3;
  }

  .footer-body {
    /* display */
    padding-right: 3em;
  }

  .footer-links {
    /* display */
    display: flex;
    height: 8em;

    /* font */
    text-align: left;

    /* flex */
    flex-flow: column wrap
  }
}

@media (min-width: 590px) {
  .footer-links { height: 7em; }
}

@media (min-width: 768px)  {
  .footer-row { display: flex; }

  .footer-logo { min-width: 330px; }

  .footer-links { height: 8em; }
}

@media (min-width: 935px) {
  .footer-links { height: 7em; }
}
