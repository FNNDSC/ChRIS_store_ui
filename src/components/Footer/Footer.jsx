import React from 'react';
import styles from './Footer.module.css';
import chrisLogo from '../../assets/img/chris_logo-white.png';

const Footer = () => (
  <div className={styles.footer}>
    <div className={styles.footerRow}>
      <div className={styles.footerLogo}>
        <img src={chrisLogo} alt="ChRIS Plugin Store" />
      </div>
      <div className={styles.footerBody}>
        <div className={styles.footerDesc}>
          ChRIS is developed by Boston Children
          &apos;
          s Hospital in partnership
          with Red Hat, the Massachusetts Open Cloud (MOC), and Boston
          University.
        </div>
        <div className={styles.footerLinks}>
          <div className={styles.footerLink}>
            <a href="https://chrisproject.org">About ChRIS</a>
          </div>
          <div className={styles.footerLink}>
            <a href="/quickstart">Submit your plugin</a>
          </div>
          <div className={styles.footerLink}>
            <a href="mailto:dev@babyMRI.org?subject=Contact from ChRIS Store">
              Contact Us
            </a>
          </div>
          <div className={styles.footerLink}>
            <a href="https://github.com/FNNDSC/ChRIS_store_ui/issues/new">
              Report an Issue
            </a>
          </div>
          <div className={styles.footerLink}>
            <a href="https://github.com/FNNDSC/ChRIS_store_ui">
              Contribute
            </a>
          </div>
          <div className={styles.footerLink}>
            <a href="mailto:dev@babyMRI.org?subject=ChRIS Store Plugin Request">
              Request a plugin
            </a>
          </div>
          {/* <div className={styles.footerLink}>Help</div>
          <div className={styles.footerLink}>Privacy Policy</div>
          <div className={styles.footerLink}>Terms & Conditions</div> */}
        </div>
      </div>
    </div>
    <div className={styles.footerCopyright}>
      Â© 2018 - {new Date().getFullYear()} Boston Children&apos;s Hospital, Red
      Hat, Massachusetts Open Cloud, Boston University. All rights reserved.
    </div>
  </div>
);

export default Footer;
