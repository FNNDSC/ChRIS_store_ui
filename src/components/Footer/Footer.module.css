.footer {
  /* color */
  background: #292e34;
  background: var(--pf-black-900);
  /* background: linear-gradient(#292e34, #030303);
  background: linear-gradient( var(--pf-black) 25%, var(--pf-black-900)); */
  color: #fafafa;
  color: var(--pf-black-100);

  /* display */
  padding: 5em 1em;
  padding-bottom: 8em;

  /* position */
  position: relative;

  /* font */
  font-size: 1.4em;
}

.footerRow {
  /* display */
  width: 100%;
  max-width: 1200px;
  margin: 0 auto !important;
}

.footerHeader {
  /* display */
  padding: 0.4em 1em;

  /* font */
  font-size: 1.3em;
  line-height: 1.3;
  white-space: pre-line;
}

.footerHeader > strong {
  font-size: 1.2em;
}

.footerLogo {
  /* display */
  padding-bottom: 2em;

  /* flex */
  flex: 1;
  flex-shrink: 0;
}

.footerLogo > img {
  /* display */
  width: 95%;
  max-width: 358.8px; /* img initial width (598px) * (3/5) */
  height: auto;
}

.footerBody {
  /* flex */
  flex: 2;
}

.footerDesc {
  font-weight: 300;
}

.footerLinks {
  /* display */
  margin-top: 2em;

  /* font */
  text-align: center;
}

.footerLink {
  /* display */
  padding-top: 0.3em;

  /* font */
  font-weight: 600;
}

.footerLink > a {
  color: #fafafa;
}

.footerLink > a:hover {
  text-decoration: none;
  color: #39a5dc;
}

.footerCopyright {
  /* position */
  position: absolute;
  bottom: 2em;
  left: 0;
  right: 0;

  /* font */
  text-align: center;
  font-size: 0.8em;
}

/* ============================== */
/* ---------- RESPONSIVE -------- */
/* ============================== */

@media (min-width: 480px) {
  .footerHeader {
    /* display */
    padding: 0.35em 0.8em;

    /* font */
    font-size: 1.4em;
    line-height: 1.3;
  }

  .footerBody {
    /* display */
    padding-right: 3em;
  }

  .footerLinks {
    /* display */
    display: flex;
    height: 8em;

    /* font */
    text-align: left;

    /* flex */
    flex-flow: column wrap;
  }
}

@media (min-width: 590px) {
  .footerLinks {
    height: 7em;
  }
}

@media (min-width: 768px) {
  .footerRow {
    display: flex;
  }

  .footerLogo {
    min-width: 330px;
  }

  .footerLinks {
    height: 8em;
  }
}

@media (min-width: 935px) {
  .footerLinks {
    height: 7em;
  }
}
