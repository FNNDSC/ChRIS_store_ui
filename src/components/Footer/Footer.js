import React from 'react';
import './Footer.css';
import chrisLogo from '../../assets/img/chris_logo-white.png';

const Footer = () => (
  <div className="footer">
    <div className="footer-row">
      <div className="footer-logo">
        <img src={chrisLogo} alt="ChRIS Plugin Store" />
      </div>
      <div className="footer-body">
        <div className="footer-desc">
          ChRIS is developed by Boston Children{"'"}s Hospital in partnership with
          Red Hat, the Massachusetts Open Cloud (MOC), and Boston University.
        </div>
        <div className="footer-links">
          <div className="footer-link">About ChRIS</div>
          <div className="footer-link">Submit your plugin</div>
          <div className="footer-link">Contact Us</div>
          <div className="footer-link">Report an Issue</div>
          <div className="footer-link">Request a plugin</div>
          <div className="footer-link">Help</div>
          <div className="footer-link">Privacy Policy</div>
          <div className="footer-link">Terms & Conditions</div>
        </div>
      </div>
    </div>
    <div className="footer-copyright">
      Â© 2018 Boston Children{"'"}s Hospital, Red Hat, Massachusetts Open Cloud,
      Boston University. All rights reserved.
    </div>
  </div>
);

export default Footer;
