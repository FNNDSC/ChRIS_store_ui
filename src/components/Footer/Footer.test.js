import React from 'react';
import { shallow } from 'enzyme';
import Footer from './Footer';

describe('Footer', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<Footer />);
  });

  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should render a div with class footer', () => {
    expect(wrapper.find('div.footer')).toHaveLength(1);
  });

  it('should render a footer-row div inside the footer div', () => {
    expect(wrapper
      .find('div.footer')
      .find('div.footer-row'))
      .toHaveLength(1);
  });

  it('should render a footer-logo div inside row div', () => {
    expect(wrapper
      .find('div.footer')
      .find('div.footer-logo'))
      .toHaveLength(1);
  });

  it('should render an img element inside footer-img div', () => {
    expect(wrapper
      .find('div.footer-logo')
      .find('img'))
      .toHaveLength(1);
  });

  it('should render img logo with alt text "ChRIS Plugin Store"', () => {
    expect(wrapper
      .find('div.footer-logo')
      .find('img')
      .prop('alt'))
      .toEqual('ChRIS Plugin Store');
  });

  it('should render a footer-body div inside row div', () => {
    expect(wrapper
      .find('div.footer')
      .find('div.footer-body'))
      .toHaveLength(1);
  });

  it('should render a footer-desc div inside footer-body div', () => {
    expect(wrapper
      .find('div.footer-body')
      .find('div.footer-desc'))
      .toHaveLength(1);
  });

  it('should render a footer-links div inside footer-body div', () => {
    expect(wrapper
      .find('div.footer-body')
      .find('div.footer-links'))
      .toHaveLength(1);
  });

  it('should render 8 footer-link divs inside footer-links div', () => {
    expect(wrapper
      .find('div.footer-links')
      .find('div.footer-link'))
      .toHaveLength(8);
  });

  it('should render footer-copyright inside footer div', () => {
    expect(wrapper
      .find('.footer')
      .find('div.footer-copyright'))
      .toHaveLength(1);
  });
});
