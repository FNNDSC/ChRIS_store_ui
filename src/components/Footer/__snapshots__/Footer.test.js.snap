// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer should render correctly 1`] = `ShallowWrapper {}`;
